tests/set/assert.h

```diff
@@ -0,0 +1,4 @@
+#ifdef NDEBUG
+#undef NDEBUG
+#endif
+#include <assert.h>
```

Comment:
Lose this, and have the makefiles remove -DNDEBUG for the test .c files's `${DFLAGS/CFLAGS.path/to/file.c}` instead.